### DISCUSSION TITLE
source code macro VM_IMAGE_SIZE

### DISCUSSION THREAD

**Discussion by chchc086** [bao-project/bao-hypervisor#163] [General]
#define VM_IMAGE_SIZE(img_name) ((size_t)&_##img_name##_vm_size)
why add "&" to access  ...vm_size value，I think its the address of  ...vm_size

**Comment by josecm** [comment]
`...vm_size` is defined by the assembly code in the macro by setting the value of the symbol, not by placing the value in memory and tagging that location with `..vm_size`. Therefore you need to take "the address", in truth, the value of the symbol, using the & operator. Hope this helps.

**Reply by sali3210** [reply]
''by setting the value of the symbol, not by placing the value in memory and tagging that location with''
could you explain this a little bit more in-depth on how I can tell in which way the symbol defined.

**Reply by chchc086** [reply]
Thanks for your response！
For the macro definition in assembly code that sets the value of a symbol, using the & operator to obtain the value can be understood as a C syntax rule, right?

**Reply by chchc086** [reply]
Or could you please tell me where I can find the answers in the official documentation?
